DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/CourseDetails/EditCourseDetailsViewModel.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.CourseDetails
+{
+    using DigitalLearningSolutions.Data.Models.Courses;
+
+    public class EditCourseDetailsViewModel : EditCourseDetailsFormData
+    {
+        public EditCourseDetailsViewModel() { }
+
+        public EditCourseDetailsViewModel(CourseDetails courseDetails)
+        {
+            CustomisationId = courseDetails.CustomisationId;
+            CentreId = courseDetails.CentreId;
+            ApplicationId = courseDetails.ApplicationId;
+            CustomisationName = GetPartOfCustomisationName(courseDetails.CustomisationName, 0)!;
+            CustomisationNameSuffix = GetPartOfCustomisationName(courseDetails.CustomisationName, 1);
+            PasswordProtected = !string.IsNullOrEmpty(courseDetails.Password);
+            Password = courseDetails.Password;
+            ReceiveNotificationEmails = !string.IsNullOrEmpty(courseDetails.NotificationEmails);
+            NotificationEmails = courseDetails.NotificationEmails;
+            PostLearningAssessment = courseDetails.PostLearningAssessment;
+            IsAssessed = courseDetails.IsAssessed;
+            DiagAssess = courseDetails.DiagAssess;
+            TutCompletionThreshold = courseDetails.TutCompletionThreshold.ToString();
+            DiagCompletionThreshold = courseDetails.DiagCompletionThreshold.ToString();

```

Comment:
Population of this stuff should probably go in a protected constructor on the FormData that takes a CourseDetails

Reply:
done!